src/ol/source/GeoTIFF.js

```diff
@@ -471,7 +447,7 @@ class GeoTIFFSource extends DataTile {
       for (let imageIndex = 0; imageIndex < imageCount; ++imageIndex) {
         const image = images[imageIndex];
         const nodataValue = image.getGDALNoData();
-        metadata[sourceIndex][imageIndex] = image.getGDALMetadata();
+        metadata[sourceIndex][imageIndex] = image.getGDALMetadata(0);
```

Comment:
This was a breaking change in geotiff@2: `GeoTIFFImage#getGDALMetadata()` now has to be called with a sample index, otherwise only metadata that is not associated with a sample will be returned. In v1, we got the metadata for the last sample, now we get the metadata for the first sample (`0`).

I *think* what we actually want is the metadata for each `wantedSample`, but making such a change would be more involved and probably deserve a separate pull request.

Reply:
Yeah, we could add another dimension to the values we store.  We could also wait until we run into a case where the nodata values are different per-band.  Curious to know if people do that.

Reply:
I'd also say we eait until we run into such a case. Just mentioned this breaking change here as a reminder that we currently assume all bands to have the same min/max values, like we and geotiff.js do for nodata.